### PR TITLE
chore: spfx scaffolding related error message

### PR DESCRIPTION
**Pull request by yuqizhou77**
- error if installing latest package failed

image

- error if running yeoman sharepoint generator failed
image

- TODO: content of help doc which will be in a separate PR
- TODO: set up aka.ms link once this PR is checked in

### CODE REVIEW

File: docs/fx-core/spfx-help-v5.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# SPFx troubleshoot
```
Comment (yuqizhou77) [inline docs/fx-core/spfx-help-v5.md:1]: content of help doc which will be in a separate PR

File: packages/fx-core/resource/package.nls.json [hunk 2]
```diff
@@ -217,13 +217,16 @@
...
+  "plugins.spfx.error.devEnvironmentNotSetup": "SPFx development environment has not been set up correctly. Visit [Set up your SharePoint Framework development environment | Microsoft Learn](%s) to learn more.",
   "_plugins.spfx.error.devEnvironmentNotSetup.comment": "'Microsoft Learn' and 'SharePoint' are product brand names, which should not be localized. ",
   "plugins.spfx.scaffold.dependencyCheck": "Checking dependencies...",
   "plugins.spfx.scaffold.dependencyInstall": "Installing dependencies. This may take more than 5 minutes to finish.",
   "plugins.spfx.scaffold.scaffoldProject": "Generate SPFx project using Yeoman CLI",
   "plugins.spfx.scaffold.updateManifest": "Update webpart manifest",
   "plugins.spfx.GetTenantFailedError": "Cannot get tenant %s %s",
+  "plugins.spfx.error.installLatestDependencyError": "Encountered unknown issues when setting up SPFx environment in HOME/.fx folder. You could follow [Set up your SharePoint Framework development environment | Microsoft Learn](%s) to set up global SPFx environment.",
```
Comment (HuihuiWu-Microsoft) [inline packages/fx-core/resource/package.nls.json:227]: Better extract 'HOME/.fx' since it'll affect localization each time the directory changes.

File: packages/fx-core/resource/package.nls.json [hunk 3]
```diff
@@ -217,13 +217,16 @@
...
+  "_plugins.spfx.error.installLatestDependencyError.comment": "'Microsoft Learn' and 'SharePoint' are the product brand names and HOME/.fx is the folder path, which should not be localized.",
+  "plugins.spfx.error.scaffoldError": "Project creation failed. A possible reason could be from Yeoman SharePoint Generator.",
```
Comment (HuihuiWu-Microsoft) [inline packages/fx-core/resource/package.nls.json:229]: From SPFx official doc, it should be 'Yeoman generator for the SharePoint Framework' or 'SPFx generator' https://learn.microsoft.com/en-us/sharepoint/dev/spfx/yeoman-generator-for-spfx-intro

Comment (yuqizhou77) [reply packages/fx-core/resource/package.nls.json:229]: I pick the words from https://learn.microsoft.com/en-us/sharepoint/dev/spfx/web-parts/get-started/build-a-hello-world-web-part#create-a-new-web-part-project "Create a new project by running the Yeoman SharePoint Generator"

File: packages/fx-core/src/component/generator/spfxGenerator.ts [hunk 4]
```diff
@@ -167,16 +172,21 @@ export class SPFxGenerator {
...
+          destinationPath,
+          context.logProvider,
+          {
+            timeout: 2 * 60 * 1000,
+            env: yoEnv,
+          },
+          "yo",
+          ...args
+        );
+      } catch (yoError) {
```
Comment (HuihuiWu-Microsoft) [inline packages/fx-core/src/component/generator/spfxGenerator.ts:187]: If we throw YoGeneratorScaffoldError directly here, we'll have difficulties troubleshooting scaffolding error. Better throw the original error message as well.

**Review by sffamily — APPROVED** [review]

File: packages/fx-core/src/component/generator/spfxGenerator.ts (hunk 4 above)
Comment (yuqizhou77) [reply packages/fx-core/src/component/generator/spfxGenerator.ts:187]: synced offline and we will put such details in output window,

**Review by HuihuiWu-Microsoft — APPROVED** [review]

**Review by sffamily — APPROVED** [review]
